nbgrader/plugins/latesubmission.py

```diff
@@ -0,0 +1,24 @@
+from textwrap import dedent
+from traitlets import Unicode
+
+from .base import BasePlugin
+
+
+class LateSubmissionPlugin(BasePlugin):
+
+    penalty_method = Unicode(
+        'none',
+        help=dedent(
+            """
+            The method for assigning late submission penalties.
+            Predefined methods:
+                'none':
+                'zero':
+            """
+        ),
+    ).tag(config=True)
+
+    def late_submission_penalty(self, student_id, score, total_seconds_late):
+        self.log.info("Using late submission penalty method: {}".format(self.penalty_method))
+        if self.penalty_method == 'zero':
```

Comment:
Should this also check the `total_seconds_late`, and only assign the penalty is the assignment is in fact late?

Ah, I see in the preprocessor you check for that already. Well, I know it's a little redundant, but it might be good to throw an error as a safeguard if `total_seconds_late` is zero.